--- a/browser/base/content/nsContextMenu.sys.mjs
+++ b/browser/base/content/nsContextMenu.sys.mjs
@@ -12,7 +12,7 @@
   BrowserUtils: "resource://gre/modules/BrowserUtils.sys.mjs",
   BrowserWindowTracker: "resource:///modules/BrowserWindowTracker.sys.mjs",
   ContextualIdentityService:
-    "moz-src:///toolkit/components/contextualidentity/ContextualIdentityService.sys.mjs",
+    "resource://gre/modules/ContextualIdentityService.sys.mjs",
   DevToolsShim: "chrome://devtools-startup/content/DevToolsShim.sys.mjs",
   E10SUtils: "resource://gre/modules/E10SUtils.sys.mjs",
   GenAI: "resource:///modules/GenAI.sys.mjs",
@@ -97,15 +97,6 @@
   "chrome://global/skin/illustrations/no-connection.svg",
 ]);
 
-const IMAGE_ONLY_PROTOCOLS = [
-  "cached-favicon:",
-  "moz-icon:",
-  "moz-newtab-wallpaper:",
-  "moz-page-thumb:",
-  "moz-remote-image:",
-  "page-icon:",
-];
-
 export class nsContextMenu {
   /**
    * A promise to retrieve the translations language pair
@@ -696,11 +687,7 @@
       this.onImage && !this.onCompletedImage
     );
 
-    // Some protocols only return images in an image context and can no longer
-    // be loaded otherwise.
-    const mediaURL = URL.parse(this.mediaURL);
-    const isImageOnlyProtocol =
-      mediaURL && IMAGE_ONLY_PROTOCOLS.includes(mediaURL.protocol);
+
 
     // View image depends on having an image that's not standalone
     // (or is in a frame), or a canvas. If this isn't an image, check
@@ -721,16 +708,10 @@
       !this.onAudio &&
       !this.onLink &&
       !this.onTextInput;
-    this.showItem(
-      "context-viewimage",
-      (showViewImage || showBGImage) && !isImageOnlyProtocol
-    );
+    this.showItem("context-viewimage", showViewImage || showBGImage);
 
     // Save image depends on having loaded its content.
-    this.showItem(
-      "context-saveimage",
-      (this.onLoadedImage && !isImageOnlyProtocol) || this.onCanvas
-    );
+    this.showItem("context-saveimage", this.onLoadedImage || this.onCanvas);
 
     if (Services.policies.status === Services.policies.ACTIVE) {
       // When file pickers are disallowed by enterprise policy,
@@ -798,12 +779,12 @@
       }
     }
 
-    let canSetDesktopBackground =
-      haveSetDesktopBackground && (this.onLoadedImage || this.onCanvas);
-
-    this.showItem("context-setDesktopBackground", canSetDesktopBackground);
+    this.showItem(
+      "context-setDesktopBackground",
+      haveSetDesktopBackground && this.onLoadedImage
+    );
 
-    if (canSetDesktopBackground) {
+    if (haveSetDesktopBackground && this.onLoadedImage) {
       this.document.getElementById("context-setDesktopBackground").disabled =
         this.contentData.disableSetDesktopBackground;
     }
@@ -906,12 +887,15 @@
 
     this.showAndFormatSearchContextItem();
     this.showTranslateSelectionItem();
+    // GORILLA: EXCISED
+    /*
     lazy.GenAI.buildAskChatMenu(document.getElementById("context-ask-chat"), {
       browser: this.browser,
       selectionInfo: this.selectionInfo,
       showItem: this.showItem.bind(this),
       source: "page",
     });
+    */
 
     // srcdoc cannot be opened separately due to concerns about web
     // content with about:srcdoc in location bar masquerading as trusted
