--- a/netwerk/protocol/http/Http3Session.cpp
+++ b/netwerk/protocol/http/Http3Session.cpp
@@ -2,6 +2,11 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+// GORILLA v2: Surgical Telemetry Lobotomy
+#undef MOZ_TELEMETRY_REPORTING
+#define MOZ_TELEMETRY_REPORTING 0
+#define GLEAN_DISABLED 1
+
 #include "ASpdySession.h"  // because of SoftStreamError()
 #include "Http3Session.h"
 #include "Http3Stream.h"
