--- a/browser/locales/en-US/browser/aboutDialog.ftl
+++ b/browser/locales/en-US/browser/aboutDialog.ftl
@@ -1,19 +1,20 @@
 # This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
-
 aboutDialog-title =
     .title = About { -brand-full-name }
 
 releaseNotes-link = What’s new
 
 update-checkForUpdatesButton =
-    .label = Check for updates
-    .accesskey = C
+    .label =
+        Check for updates
+        .accesskey = C
 
 update-updateButton =
-    .label = Restart to Update { -brand-shorter-name }
-    .accesskey = R
+    .label =
+        Restart to Update { -brand-shorter-name }
+        .accesskey = R
 
 update-checkingForUpdates = Checking for updates…
 settings-update-checking-for-updates =
@@ -21,20 +22,17 @@
 
 ## Variables:
 ##   $transfer (string) - Transfer progress.
-
 settings-update-downloading-2 =
     .label = Downloading update — { $transfer }
 aboutdialog-update-downloading = Downloading update — <label data-l10n-name="download-status">{ $transfer }</label>
 
 ##
-
 update-applying = Applying update…
 settings-update-applying =
     .label = Applying update…
 
-update-failed = Update failed. <label data-l10n-name="failed-link">Download the latest version</label>
-update-failed-main =
-    Update failed. <a data-l10n-name="failed-link-main">Download the latest version</a>
+update-failed = Update failed. <label data-l10n-name="failed-link">Gorilla Download the latest version</label>
+update-failed-main = Update failed. <a data-l10n-name="failed-link-main">Gorilla Download the latest version</a>
 
 update-policy-disabled = Updates disabled by your organization
 settings-update-policy-disabled =
@@ -51,7 +49,6 @@
 
 ## Variables:
 ##   $displayUrl (String): URL to page with download instructions. Example: www.mozilla.org/firefox/nightly/
-
 aboutdialog-update-manual-with-link = Updates available at <label data-l10n-name="manual-link">{ $displayUrl }</label>
 settings-update-manual-with-link = Updates available at <a data-l10n-name="manual-link">{ $displayUrl }</a>
 
@@ -66,7 +63,6 @@
 settings-update-internal-error = Unable to check for updates due to internal error. Updates available at <a data-l10n-name="manual-link">{ $displayUrl }</a>
 
 ##
-
 # Variables:
 #   $channel (String): description of the update channel (e.g. "release", "beta", "nightly" etc.)
 aboutdialog-channel-description = You are currently on the <label data-l10n-name="current-channel">{ $channel }</label> update channel.
