--- a/browser/locales/jar.mn
+++ b/browser/locales/jar.mn
@@ -11,7 +11,6 @@
   preview/interventions.ftl                        (../components/urlbar/content/interventions.ftl)
   preview/enUS-searchFeatures.ftl                  (../components/urlbar/content/enUS-searchFeatures.ftl)
   preview/onboarding.ftl                           (../components/aboutwelcome/content/onboarding.ftl)
-  preview/genai.ftl                                (../components/genai/content/genai.ftl)
   preview/credentialChooser.ftl                    (../../toolkit/components/credentialmanagement/credentialChooser.ftl)
   browser                                          (%browser/**/*.ftl)
   preview/smartTabGroups.ftl                       (../locales-preview/smartTabGroups.ftl)
