--- a/toolkit/components/messaging-system/lib/SpecialMessageActions.sys.mjs
+++ b/toolkit/components/messaging-system/lib/SpecialMessageActions.sys.mjs
@@ -27,15 +27,11 @@
 
 ChromeUtils.defineESModuleGetters(lazy, {
   AddonManager: "resource://gre/modules/AddonManager.sys.mjs",
-  AIWindow:
-    // eslint-disable-next-line mozilla/no-browser-refs-in-toolkit
-    "moz-src:///browser/components/aiwindow/ui/modules/AIWindow.sys.mjs",
   // eslint-disable-next-line mozilla/no-browser-refs-in-toolkit
   CustomizableUI:
     "moz-src:///browser/components/customizableui/CustomizableUI.sys.mjs",
   ExperimentAPI: "resource://nimbus/ExperimentAPI.sys.mjs",
   FxAccounts: "resource://gre/modules/FxAccounts.sys.mjs",
-  GenAI: "resource:///modules/GenAI.sys.mjs",
   IPProtection:
     // eslint-disable-next-line mozilla/no-browser-refs-in-toolkit
     "moz-src:///browser/components/ipprotection/IPProtection.sys.mjs",
@@ -970,9 +966,6 @@
       case "FXA_SIGNIN_FLOW":
         /** @returns {Promise<boolean>} */
         return this.fxaSignInFlow(action.data, browser);
-      case "FXA_AIWINDOW_SIGNIN_FLOW":
-        /** @returns {Promise<boolean>} */
-        return lazy.AIWindow.launchWindow(browser);
       case "OPEN_PROTECTION_PANEL": {
         let { gProtectionsHandler } = window;
         gProtectionsHandler.showProtectionsPopup({});
@@ -1057,11 +1050,6 @@
         window.gURLBar.searchMode = action.data;
         window.gURLBar.focus();
         break;
-      case "SUMMARIZE_PAGE": {
-        const entry = action.data ?? "message";
-        await lazy.GenAI.summarizeCurrentPage(window, entry);
-        break;
-      }
       case "OPEN_PANEL": {
         let { anchor_id, widget_id, panel_id, fallback_to_app_menu } =
           action.data;
